Lamps, Bits, Bobs and Doodads.

BloodyMan opened the pull request:
:cl: Cajoes
rscadd: Added a lube variant called oil, comes in jugs of 100 units, dispensable from robotics vendor.
rscadd: Added beer and ale cans to the alcohol vendor.
rscadd: Added lava lamps and office toys to the misc. supply menu.
rscadd: Added fish meat and a big bag of salt to the kitchen.
tweak: Changed contents of some drinks.
/:cl:

BloodyMan committed:
Doodads: Final version.

Lava Lamps

fixes

format fix

Added var definition fix.

fix fixes.

Added Desktop toys and a fan

added dipping bird toy

format fix

Added some plants I made yons ago.

wrote desk instead of desc on the desk toys.

container for the oil, desk toy crate

crate conflict resolved

added beers in a can

Beer and Toy Fixes

tab fix

Toy Fixes, Drink Fix, Lamp Def Fix

Cannibalizing code

Fix fix fix

Fixes the on-off toggle.. I think?

Final fix. Fuck everything.

Changed for quote accuracy

Centering sprites

Quote fix

Added unused assets to vendors.

Toggle def fixed.

forgot the commas

fish and salt and tweaks

polish

Tweak Pass

doot

polish

carp fixes

spellcheck

Sprite fixes, visa added

EcklesFire commented:
Does oil function like lube does?

BloodyMan commented:
On paper, yes?

babydoll commented on code/game/objects/items/devices/flashlight.dm:
```diff
@@ -390,4 +390,52 @@
+	color = "#ff0000"
+
+/obj/item/device/flashlight/lavalamp/blue
+	name = "A kitchy blue decorative light"
```
Should be `desc`, ditto on all the others

babydoll commented on code/game/objects/items/devices/flashlight.dm:
```diff
@@ -390,4 +390,52 @@
+/obj/item/device/flashlight/lavalamp/red
+	desc = "A kitchy red decorative light."
+	icon_state = "redlamp"
+	color = "#ff0000"
```
Why are you setting the `color` var here?

BloodyMan replied:
So lamp... light is different?

babydoll replied:
`color` determines the colour of the icon, `light_color` determines the colour of the light. The glowsticks use `color` since they have a greyscale icon.

BlueNexus commented on code/game/objects/items/devices/flashlight.dm:
```diff
@@ -390,4 +390,80 @@
+
+/obj/item/device/flashlight/desk/fan
+	name = "office fan"
+	desk = "Your greatest fan."
```
desc, not desk

BlueNexus commented on code/game/objects/items/devices/flashlight.dm:
```diff
@@ -390,4 +390,80 @@
+
+/obj/item/device/flashlight/desk/officetoy
+	name = "office toy"
+	desk = "A generic microfusion powered office desk toy. Only generates magnetism and ennui."
```
ditto

BlueNexus commented on code/game/objects/items/devices/flashlight.dm:
```diff
@@ -390,4 +390,80 @@
+
+/obj/item/device/flashlight/desk/dippingbird
+	name = "dipping bird toy"
+	desk = "A ancient human bird idol, worshipped by clerks and desk jockeys."
```
ditto

BlueNexus commented on code/game/objects/items/devices/flashlight.dm:
```diff
@@ -390,4 +390,80 @@
+
+/obj/item/device/flashlight/desk
+	name = "\improper This does not exist"
+	desk = "A object that does not exist."
```
ditto

babydoll commented on code/game/objects/items/devices/flashlight.dm:
```diff
@@ -390,4 +390,80 @@
+
+//Desk Doodads that don't neccessarily emit light, but I don't know where else to put them and flashlights already have a on-off toggle var.
+
+/obj/item/device/flashlight/desk
```
These all should probably be subtypes of `obj/item/toy` and not flashlights. See https://github.com/Baystation12/Baystation12/blob/dev/code/game/objects/items/toys.dm

BloodyMan replied:
But do toys have on-off vars?

CrimsonShrike replied:
BloodyMan, if they're subtypes of flashlights you're going to be able to aim them at people's eyes to tell if they're high amongst other things.

Anton-Kr commented on code/modules/reagents/Chemistry-Reagents/Chemistry-Reagents-Food-Drinks.dm:
```diff
@@ -1123,7 +1123,7 @@
 /datum/reagent/ethanol/beer
 	name = "Beer"
 	description = "An alcoholic beverage made from malted grains, hops, yeast, and water."
-	taste_description = "piss water"
```
:(

babydoll commented on code/game/objects/items/devices/flashlight.dm:
```diff
@@ -390,4 +390,52 @@
-//Lava Lamps: Because we're already stuck in the 70ies with those fax machines.
\ No newline at end of file
+//Lava Lamps: Because we're already stuck in the 70ies with those fax machines.
+/obj/item/device/flashlight/lavalamp
```
`/obj/item/device/flashlight/lamp/lavalamp`, as discussed on discord

babydoll commented on code/game/objects/items/toys.dm:
```diff
@@ -842,4 +844,38 @@
+
+	var on = 0
+
+/obj/item/toy/desk/proc/update_icon()
```
I still don't think you should be doing anything other than updating the icon in update_icon.

Also, this should be `/obj/item/toy/desk/update_icon()`

babydoll commented on code/game/objects/items/toys.dm:
```diff
@@ -842,4 +844,38 @@
+    icon_state = "[initial(icon_state)][state]-[on]"
+
+/obj/item/toy/desk/newtoncradle
+	name = "\improper Newton's Cradle"
```
`\improper Newton's cradle`

BloodyMan commented:
[Screenshot](https://www.dropbox.com/s/8e91q0mk9ubq3jx/scrnshot1%20%282%29.png?dl=0)

ghost commented:
Add these new cans to Booze-o-Mats, maybe? You could also put a couple oil jugs in borg recharge stations in the maintenance, just for fun.

babydoll commented on code/game/objects/items/toys.dm:
```diff
@@ -842,4 +844,56 @@
+	desc = "A object that does not exist. Parent Item"
+	icon = 'icons/obj/toy.dmi'
+
+	action_button_name = "Toggle Flashlight"
```
Why do these need action buttons?

babydoll replied:
Also, these aren't flashlights.

BloodyMan replied:
On and off animations. Very detailed office desk toys.

babydoll commented on code/game/objects/items/toys.dm:
```diff
@@ -842,4 +844,56 @@
+
+/obj/item/toy/desk/Initialize()
+	. = ..()
+	update_icon()
```
Why are we updating the icon here?

BloodyMan replied:
I do not know. I am afraid to do any more alterations because all attempts have resulted in the objects breaking.

babydoll commented on code/modules/reagents/Chemistry-Reagents/Chemistry-Reagents-Food-Drinks.dm:
```diff
@@ -1131,6 +1131,17 @@
+	nutriment_factor = 1
+
+	glass_name = "beer"
+	glass_desc = "A freezing pint of beer"
```
Could be in any number of different glasses that do not contain a pint

BloodyMan replied:
That was the original description.

babydoll replied:
The original description is wrong, too

babydoll commented on code/modules/reagents/Chemistry-Reagents/Chemistry-Reagents-Food-Drinks.dm:
```diff
@@ -1131,6 +1131,17 @@
 	glass_name = "beer"
 	glass_desc = "A freezing pint of beer"
 
+/datum/reagent/ethanol/goodbeer
```
Where is this used?

BloodyMan replied:
For now, in canned beer.

babydoll replied:
Make this a subtype of beer and override taste_description instead of this copy/paste

babydoll commented on code/modules/reagents/Chemistry-Reagents/Chemistry-Reagents-Other.dm:
```diff
@@ -300,6 +300,16 @@
 	if(volume >= 1)
 		T.wet_floor(80)
 
+/datum/reagent/lube/oil // TODO: Robot Overhaul in general
```
Add it when you plan on using it

BloodyMan replied:
Fixed.

babydoll reviewed (changes requested):
Don't add things that aren't being used - if you want to add a different kind of beer, give it a premium name like the different whiskey and add it _somewhere._

e: For clarification, I am requesting the things that were added but aren't used are either removed and saved for a later PR, or implemented somewhere where they're obtainable in-game.

babydoll commented:
Also, changelog meaning, something that will show up in the changelog in game.

```
:cl: yourname
rscadd: A thing you added
tweak: A thing you changed
rscdel: A thing you removed
:/cl:

BloodyMan commented:
... Where do I put that?

PsiOmegaDelta commented:
In the main post of the PR.

PsiOmegaDelta commented on code/game/objects/items/toys.dm:
```diff
@@ -842,4 +844,56 @@
+		icon_state = "[initial(icon_state)]"
+
+/obj/item/toy/desk/attack_self(mob/user)
+	if(!isturf(user.loc))
```
I don't see why this is a requirement.

BloodyMan replied:
I just copy-pasted the flashlight code and removed all lines with flashlight in them. I am afraid to alter it further lest it break its function.

babydoll replied:
Potentially because action buttons call attack_self? Action button is gone so you can get rid of this check.

babydoll commented on code/game/objects/items/toys.dm:
```diff
@@ -842,4 +844,56 @@
+	desc = "A object that does not exist. Parent Item"
+	icon = 'icons/obj/toy.dmi'
+
+	action_button_name = "Toggle Toy"
```
These do not need an action button. Those are for convenience for things like flashlights and HUDs so you don't need to unequip them to turn them on/off.

babydoll commented on code/game/objects/items/toys.dm:
```diff
@@ -842,4 +844,56 @@
+/obj/item/toy/desk
+	name = "desk toy master"
+	desc = "A object that does not exist. Parent Item"
+	icon = 'icons/obj/toy.dmi'
```
Superfluous, /toy `icon` is already set to the same

babydoll commented on code/datums/supplypacks/miscellaneous.dm:
```diff
@@ -1,6 +1,29 @@
 /decl/hierarchy/supply_pack/miscellaneous
 	name = "Miscellaneous"
 
+/decl/hierarchy/supply_pack/miscellaneous/llamps
```
Consider making this give you 2-3 random lamps instead of every possible one each time - look at the costumes crate for an example

BloodyMan replied:
Done

babydoll commented on code/game/objects/items/toys.dm:
```diff
@@ -842,4 +844,54 @@
+	if(on && activation_sound)
+		playsound(src.loc, activation_sound, 75, 1)
+	update_icon()
+	user.update_action_buttons()
```
Action button remnant

babydoll commented on code/modules/reagents/reagent_containers/food/snacks.dm:
```diff
@@ -621,11 +621,18 @@
 	filling_color = "#ffdefe"
 	center_of_mass = "x=17;y=13"
 
+	New()
```
All new pathing must be absolute

babydoll commented on code/modules/reagents/reagent_containers/food/snacks.dm:
```diff
@@ -621,11 +621,18 @@
+		reagents.add_reagent(/datum/reagent/nutriment/protein, 3)
+		src.bitesize = 6
+
+/obj/item/weapon/reagent_containers/food/snacks/carpmeat/space
```
You've done this in reverse for some reason. Make the new thing the subtype.

babydoll commented on code/modules/reagents/reagent_containers/food/drinks/bottle.dm:
```diff
@@ -478,3 +478,14 @@
+
+//Probably not the right place for it, but no idea where else to put it without making a brand new DM and slogging through making vars from scratch.
+
+/obj/item/weapon/reagent_containers/food/drinks/bottle/oiljug
```
Needs `isGlass` set to 0

BloodyMan replied:
done

babydoll commented on code/modules/reagents/reagent_containers/food/condiment.dm:
```diff
@@ -215,3 +215,18 @@
+
+/obj/item/weapon/reagent_containers/food/condiment/salt/New()
+	..()
+	reagents.add_reagent(/datum/reagent/sodiumchloride, 250)
```
Putting 250 of a reagent in a container that holds 50

babydoll commented on code/game/objects/structures/crates_lockers/closets/secure/freezer.dm:
```diff
@@ -21,7 +22,8 @@
 	return list(
-		/obj/item/weapon/reagent_containers/food/snacks/meat/monkey = 10
+		/obj/item/weapon/reagent_containers/food/snacks/meat/beef = 8,
+		/obj/item/weapon/reagent_containers/food/snacks/carpmeat = 4
```
You probably want this to be the safe variant you created below

BloodyMan replied:
Nah, chefs gotta learn somehow.

babydoll commented on code/modules/reagents/reagent_containers/food/condiment.dm:
```diff
@@ -215,3 +215,19 @@
+	icon_state = "salt"
+	item_state = "flour"
+	randpixel = 10
+	volume = 500
```
Needs a w_class set, this is the same size as a salt shaker as far as the game is concerned

BloodyMan replied:
set it to large for now.

babydoll commented on code/game/objects/items/toys.dm:
```diff
@@ -842,4 +847,50 @@
+
+/obj/item/toy/desk/Initialize()
+	. = ..()
+	update_icon()
```
None of these begin `on` so there's no reason to `update_icons` here, and it's the only thing in `Initialize` so you can get rid of all of it.

babydoll commented on code/modules/reagents/reagent_containers/food/snacks.dm:
```diff
@@ -619,14 +619,21 @@
+/obj/item/weapon/reagent_containers/food/snacks/carpmeat/safe
 	name = "carp fillet"
-	desc = "A fillet of spess carp meat."
+	desc = "A fillet of carp meat. Syntethized from ancient earth genetic archives."
```
synthesized ... Earth

babydoll commented on code/modules/reagents/reagent_containers/food/snacks.dm:
```diff
@@ -619,14 +619,21 @@
 		bitesize = 1
 
-/obj/item/weapon/reagent_containers/food/snacks/carpmeat
+/obj/item/weapon/reagent_containers/food/snacks/carpmeat/safe
```
Listing the subtype before the master

babydoll commented on code/modules/reagents/reagent_containers/food/snacks.dm:
```diff
@@ -619,14 +619,21 @@
+		src.bitesize = 6
+
+/obj/item/weapon/reagent_containers/food/snacks/carpmeat
+	desc = "A fillet of spess carp meat."
```
space carp

BloodyMan replied:
original spelling.

babydoll replied:
The original spelling is wrong

babydoll commented on code/modules/reagents/reagent_containers/food/snacks.dm:
```diff
@@ -620,16 +620,23 @@
+/obj/item/weapon/reagent_containers/food/snacks/carpmeat/safe
 	name = "carp fillet"
-	desc = "A fillet of spess carp meat."
+	desc = "A fillet of carp meat. Syntethized from ancient Earth genetic archives."
```
synthesized

babydoll commented:
What'd you add to bureaucracy.dmi and plants.dmi? I don't see anything in code that might use those files.

BloodyMan commented:
plants.dmi got a few more potted plants, forgot to assign them to existing plant-code. Basicly the precursors to todays ubiquotous "make workplace bearable" initative. Lemme dig back into the code a bit.

I completely forgot what I added to bureocracy.

BloodyMan commented:
Okay, added the plants to new plant entries.

Apprently I left in the bureaucracy.dmi the following.
* updated green pen icon to bring it in line with the shade of the other two primary colored pens.
* Travel visa
* roll of red tape
* jug of liquid paper
I have added a travel visa to the items, would have to tinker with loadouts.dm if you want them ingame. Shall I proceed with this?

babydoll commented:
It's your PR, you can do whatever you like, but please do not include any icons that you aren't **immediately** using.

BloodyMan commented:
Have trimmed all ancillary crap as of the final version.

PsiOmegaDelta commented on code/modules/reagents/reagent_containers/food/condiment.dm:
```diff
@@ -215,3 +215,20 @@
+	item_state = "flour"
+	randpixel = 10
+	volume = 500
+	w_class = ITEM_SIZE_LARGE
```
I'll just note that large items include, but are not limited to:
* 1 meter long pipes
* Flamethrowers
* Toolboxes
* Assault rifles
* Gas tanks
